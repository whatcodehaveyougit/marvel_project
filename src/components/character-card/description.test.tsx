import { render } from "@testing-library/react";
import * as character from "../../testData/character-object.json";
import Description from "./description";
import { describe, it, expect, jest } from "@jest/globals";

// Remove the unnecessary console.log statement
// React.useRef = () => ({ current: { clientHeight: 100, scrollHeight: 200 } });

describe("Description", () => {
  it("renders correctly with another character and useRef being mocked", () => {
    // reactMock.useRef = { current: { clientHeight: 100, scrollHeight: 200 } };
    // const useRefSig = jest.fn();
    jest.mock("react", () => {
      // const originReact = jest.requireActual('react');
      return {
        // ...originReact,
        hello: "world",
        useRef: { current: { clientHeight: 100, scrollHeight: 200 } }, // Not working
      };
    });

    const { asFragment } = render(<Description character={character} />);

    expect(asFragment()).toMatchSnapshot();
  });
});

// Problem

// Cannot mock useRef in the test file and the HTML components are not rendered with
// All the attributes and values as normal so it does't go through the whole if statement
// So doesn't test the whole file
// I think it is useRef that I need to mock but not 100% sure.,.
