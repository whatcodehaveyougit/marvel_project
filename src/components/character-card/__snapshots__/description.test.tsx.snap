// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders correctly with another character and useRef being mocked 1`] = `
<DocumentFragment>
  <div
    class="container-1011334 max-lines"
  >
    <div>
      <p
        class="text-gray-500"
      >
        No superpowers ;(
      </p>
    </div>
  </div>
</DocumentFragment>
`;
